src/components/AccessForm.vue

```diff
@@ -73,24 +79,66 @@ export default defineComponent({
             trigger: "change"
           }
         ]
-      }
+      },
+      loading: false
     };
   },
+  computed: {
+    ...mapState(["ourSecretKey"]),
+    ...mapGetters(["enclavePublicKey"])
+  },
   methods: {
+    ...mapActions(["sealBox"]),
     onSubmit(formName: string) {
       (this.$refs[formName] as typeof elForm).validate(
         async (valid: boolean) => {
           if (valid) {
-            const enclavePublicKey = this.$store.getters.enclavePublicKey;
+            this.loading = true;
             // Get the unproxied form data before encrypting, for clarity.
             const data = Object.assign({}, this.form);
-            await encryptJson(data, enclavePublicKey);
+            await this.sealBox(data).then(async messageData => {
+              if (messageData) {
+                await this.postEncryptedBox(messageData);
+              }
+            });
           } else {
             console.log("error submit!!");
             return false;
           }
         }
       );
+    },
+    async postEncryptedBox(messageData: MessageData) {
+      const payload = {
+        metadata: {
+          nonce: messageData.nonce,
+          uploader_pub_key: messageData.ourPublicKey
+        },
+        payload: messageData.ciphertext
+      };
+      await this.axios
+        .post("https://rtc-data.registree.io/auth/tokens", payload)
+        .then(async result => {
```

Comment:
We should probably avoid mixing both async/await and `Promise` syntax?

Standardising on async/await probably makes most sense here, for better tracebacks?